Tools/PowershellModule/doc/WingetRestSource.md

```diff
@@ -1,575 +1,176 @@
-    PS C:\> $RestAPI = "C:\Projects\winget-cli-restsource\WinGet.RestSource.Functions.zip"
-    PS C:\> Connect-AzAccount -SubscriptionName "Contoso Azure Subscription"
-    PS C:\> Publish-AzWebApp -ArchivePath $RestAPI -ResourceGroupName "WinGet_RestSource" -Name "contoso-function-demo" -Force
-    ```
+# Create a Windows Package Manager REST source
+
+This section provides guidance on how to create a REST source for the Windows Package Manager. ISVs or Publishers may host and manage a REST source if they would like full control of the packages available in a source. An independently hosted source may choose to expose the read endpoints publicly, restrict access to specific IP address via the addition of a traffic shaping module or restrict access by Microsoft Entra Id authentication.  The basic setups configured by the cmdlets and examples result in a source that is publicly readable but requires an authorization key to manage. 
+
+Windows Package Manager offers a comprehensive package manager solution including a command line tool and a set of services for installing applications. For more general package submission information, see [submit packages to Windows Package Manager](https://learn.microsoft.com/windows/package-manager/package/).
+
+## Manage Windows Package Manager REST source with PowerShell
+
+To simplify the management and interaction with the Windows Package Manager REST source, the `Microsoft.WinGet.RestSource` PowerShell module has been made available. This PowerShell module provides cmdlets that enable you to Add, Find, Remove and Get package manifests from your Windows Package Manager REST source, as well as stand up a new Windows Package Manager REST source in Azure.
+
+The following steps are for managing a Windows Package Manager REST source with PowerShell:
+
+1. Download and install the `Microsoft.WinGet.RestSource` PowerShell Module.
+2. Automate the creation of a Windows Package Manager REST source.
+3. Publish Package Manifests to the Windows Package Manager REST source.
+4. Find Package Manifests from Windows Package Manager REST source.
+5. Retrieve published Package Manifests from the Windows Package Manager REST source.
+6. Remove published Package Manifests from the Windows Package Manager REST source.
+
+### PowerShell Pre-Requisites
+
+**The `Microsoft.WinGet.RestSource` PowerShell module requires PowerShell Core version 7.4 or later.**
+
+Before getting started with the Windows Package Manager REST source with PowerShell, here are a few recommended steps you should complete before using the PowerShell module to ensure you can successfully stand up a new Windows Package Manager REST source in Azure.
+
+1. Open PowerShell as an Administrator
+2. Set the PowerShell execution policies
+    ```Powershell
+    PS C:\> Set-ExecutionPolicy -ExecutionPolicy RemoteSigned -Scope CurrentUser
+    ```
+4. Install the Azure Az module 
+    ```Powershell
+    PS C:\> Install-Module -Name Az -AllowClobber

```

Comment:
Do the same for this. Using Install-PSResource over Install-Module for Az is way faster :)

Reply:
Does this work in Windows PowerShell also?

Reply:
If they have the `Microsoft.PowerShell.PSResourceGet` module installed, yes. Otherwise, you can give both options something like:

```powershell
# Using Windows PowerShell
PS C:\> Install-Module -Name Az -AllowClober 

# Using PowerShell 7.4+
PS C:\> Install-PSResource -Name Az 
```

Reply:
To Denelon, the rest source module uses .net 8 built dlls, .net 8 is built in with PowerShell Core 7.4 or later, otherwise some manual workarounds is needed. In the psd1, I set to Core 7.4.
To Gijsreyn, I'm just following instructions listed in PS Gallery. I don't feel to give instructions not listed in PSGallery.

Reply:
I think there are ;). Same would be for Az. Performance with Install-PSResource is way faster compared to Install-Module.

![image](https://github.com/user-attachments/assets/a46328f2-85ec-4dff-b6fb-99f5ce1914ea)

Reply:
I meant the Az one, I updated the RestSource one with Install-PSResource

Reply:
Given a requirement for PowerShell 7, can we ensure Install-PSResource is installed so our implementation doesn't fail?

Reply:
Install-PSResource is included in PS Core 7.4 +.  We do not use Install-PSResource in our script, so technically our module is not installed yet if user failed to `Install-PSResource -Name Microsoft.WinGet.RestSource -Prerelease`

Reply:
I think the main complexity in my mind is the subset that work with Windows PowerShell vs. those that require PowerShell 7.

Reply:
> I think the main complexity in my mind is the subset that work with Windows PowerShell vs. those that require PowerShell 7.

We could make this module work on Windows PowerShell by building another copy of the WinGet.RestSource.PowerShellSupport dll using .net framework.